orquesta/tests/unit/conducting/test_workflow_conductor_rerun.py

```diff
@@ -0,0 +1,629 @@
+
+    def test_parallel_rerun_with_first_task(self):
+        expected_output = {
+            'var1': 'xyz',
+            'var2': 123
+        }
+
+        # 1. Prepare rerun workflow
+        conductor = self.pre_rerun_workflow(PARALLEL_WF_DEF)
+
+        # 2. Fail task1
+        # Conduct tasks.
+        self.forward_task_statuses(conductor, 'task1', [statuses.RUNNING])
+        self.forward_task_statuses(conductor, 'task3', [statuses.RUNNING])
+
+        self.forward_task_statuses(conductor, 'task1', [statuses.FAILED])
+        self.forward_task_statuses(conductor, 'task3', [statuses.SUCCEEDED])
+
+        expected_t3_ctx = {'var1': 'abc', 'var2': 123}
+        expected_txsn_ctx = {'task4__t0': expected_t3_ctx}
+        self.assertDictEqual(conductor.get_task_transition_contexts('task3', 0), expected_txsn_ctx)
+
+        # log the error
+        error = 'test parallel rerun with first task'
+        conductor.log_error(error, 'task1')
+
+        # Check the list of terminal tasks.
+        term_tasks = conductor.workflow_state.get_terminal_tasks()
+        actual_term_tasks = [t['id'] for t in term_tasks]
+        expected_term_tasks = ['task1', 'task3']
+        self.assertListEqual(actual_term_tasks, expected_term_tasks)
+        self.assertEqual(conductor.get_workflow_status(), statuses.FAILED)
+
+        # 3. Start rerun workflow
+        options = dict()
+        options['tasks'] = ['task1']
+        conductor.request_workflow_rerun(options)
+
+        # Conduct task4 and check context.
+        self.forward_task_statuses(conductor, 'task4', [statuses.RUNNING])
+        self.forward_task_statuses(conductor, 'task4', [statuses.SUCCEEDED])
+        self.assertDictEqual(conductor.get_task_transition_contexts('task4', 0), {})
+
+        # Conduct task1 and check context.
+        self.forward_task_statuses(conductor, 'task1', [statuses.RUNNING])
+        self.forward_task_statuses(conductor, 'task1', [statuses.SUCCEEDED])
+
+        expected_t1_ctx = {'var1': 'xyz', 'var2': 234}
+        expected_txsn_ctx = {'task2__t0': expected_t1_ctx}
+        self.assertDictEqual(conductor.get_task_transition_contexts('task1', 0), expected_txsn_ctx)
+
+        task_ids = ['task2']
+        task_ctxs = [expected_t1_ctx]
+        task_routes = [0]
+        self.assert_next_tasks(conductor, task_ids, task_ctxs, task_routes)
+
+        # Conduct task2 and check context.
+        self.forward_task_statuses(conductor, 'task2', [statuses.RUNNING])
+        self.forward_task_statuses(conductor, 'task2', [statuses.SUCCEEDED])
+        self.assertDictEqual(conductor.get_task_transition_contexts('task2', 0), {})
+        self.assert_next_task(conductor, has_next_task=False)
+
+        # Check the list of terminal tasks.
+        term_tasks = conductor.workflow_state.get_terminal_tasks()
+        actual_term_tasks = [t['id'] for t in term_tasks]
+        expected_term_tasks = ['task1', 'task3', 'task4', 'task2']
```

Comment:
We missed this. The term tasks should only be `task2` and `task4`.  In `request_workflow_rerun`, you have to set the original `task['term'] = False`.